### PR TITLE
NonInclusiveLanguageChecker to flag words with multiple spaces

### PR DESCRIPTION
**Pull request by binamaniar**
rdar://75552499

## Summary

The non-inclusive language checker doesn’t match against words with multiple spaces. It matches words like “whitelist” and “white list” but not “white       list”.

## Dependencies

No Known dependencies 

## Testing

Steps:

1. Create a project in Xcode, it can be a swift project
2. Create .docc folder in that.
3. have an overview.md at .docc level, i/e /docc/overview.md
4. Add non inclusive language with multiple space, such as "white   list" within that file
5. Build documentation.
6. Diagnostic to be added for the above non inclusive language.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
- [yes ] Added tests
- [ yes] Ran the `./bin/test` script and it succeeded
- [ NA] Updated documentation if necessary

### CODE REVIEW

**Comment by binamaniar** [comment]
@swift-ci please test

**Comment by Kyle-Ye** [comment]
The rendered output of your markdown context shows both "white list" and "white list" on the summary.

I checked your markdown source and got your meaning.

But I think we should update it to the following:
```
white  list
```

File: Tests/SwiftDocCTests/Checker/Checkers/NonInclusiveLanguageCheckerTests.swift [hunk 1]
```diff
@@ -31,6 +31,40 @@ class NonInclusiveLanguageCheckerTests: XCTestCase {
...
+        XCTAssertEqual(rangeTwo.upperBound.line, 2)
+        XCTAssertEqual(rangeTwo.upperBound.column, 20)
+
+        let problemThree = try XCTUnwrap(checker.problems[2])
+        let rangeThree = try XCTUnwrap(problemThree.diagnostic.range)
+        XCTAssertEqual(rangeThree.lowerBound.line, 3)
+        XCTAssertEqual(rangeThree.lowerBound.column, 5)
+        XCTAssertEqual(rangeThree.upperBound.line, 3)
+        XCTAssertEqual(rangeThree.upperBound.column, 17)
+
```
Comment (Kyle-Ye) [inline Tests/SwiftDocCTests/Checker/Checkers/NonInclusiveLanguageCheckerTests.swift:65]: Sorry for the nitpick here, but I think we should remove line 65 and update the year in the license header from `2021` to `2021-2022` in the 2 files.

**Comment by Kyle-Ye** [comment]
@swift-ci please test

**Comment by Kyle-Ye** [comment]
The failing test case is probably caused by the recently swift-markdown change. Maybe we should update with rebase to include #295 to fix it.

File: Tests/SwiftDocCTests/Checker/Checkers/NonInclusiveLanguageCheckerTests.swift [hunk 2]
```diff
@@ -31,6 +31,40 @@ class NonInclusiveLanguageCheckerTests: XCTestCase {
         XCTAssertEqual(range.upperBound.column, 16)
     }
 
+    func testMatchTermWithSpaces() throws {
+        let source = """
+# A White  listed title
+# A Black    listed title
+# A White listed title
+"""
```
Comment (franklinsch) [inline Tests/SwiftDocCTests/Checker/Checkers/NonInclusiveLanguageCheckerTests.swift:39]:
```suggestion
        # A White  listed title
        # A Black    listed title
        # A White listed title
        """
```

Would recommend aligning the end of the string here to the start column-wise to align with the style of the rest of the repo.

**Comment by binamaniar** [comment]
@swift-ci please test

**Comment by binamaniar** [comment]
@swift-ci please test

File: Tests/SwiftDocCTests/Checker/Checkers/NonInclusiveLanguageCheckerTests.swift (hunk 1 above)
Comment (binamaniar) [reply Tests/SwiftDocCTests/Checker/Checkers/NonInclusiveLanguageCheckerTests.swift:67]: Thank you for pointing that out, i updated that.

File: Tests/SwiftDocCTests/Checker/Checkers/NonInclusiveLanguageCheckerTests.swift (hunk 2 above)
Comment (binamaniar) [reply Tests/SwiftDocCTests/Checker/Checkers/NonInclusiveLanguageCheckerTests.swift]: sounds good. done 👍🏻

**Comment by binamaniar** [comment]
@swift-ci please test

**Comment by binamaniar** [comment]
@swift-ci please test

**Review by ethan-kusters — APPROVED** [review]
Thank you!

**Review by franklinsch — APPROVED** [review]

**Pull request merged**